dbt/adapters/spark_livy/livysession.py

```diff
@@ -191,11 +189,15 @@ def execute(self, sql: str, *parameters: Any) -> None:
                 # print("rows", self._rows)
                 # print("schema", self._schema)
             else:
-                self._rows = None
-                self._schema = None
+                self._rows = []
+                self._schema = []
         else:
             self._rows = None
-            self._schema = None 
+            self._schema = None
+
+            raise dbt.exceptions.raise_database_error(
+                        'Error while executing query: ' + res['output']['evalue']
```

Comment:
Did we check if this is printing the json object properly in case of an  actual error.